src/npr/ont_pipeline.py

```diff
@@ -165,14 +171,80 @@ def read_flowcell_info(config, info_dict, base_path):
             '*json'
         )
     )
+    html_file = glob.glob(
+        os.path.join(
+            flowcell_path,
+            'reports',
+            '*html'
+        )
+    )
+    if html_file:
+        print("[green]Reading info from html[/green]")
+        html_cont = ''
+        with open(html_file[0]) as f:
+            html_cont = f.read()
+        
+        info_dict["software"] = {}
+        matches = re.finditer(match_html_key_val, html_cont)
+        for match in matches:
+            name = match.group(1)
+            value = match.group(2)
+            if name == "Flow cell type":
+                info_dict["flowcell"] = value
+            elif name == "Kit type":
+                info_dict["kit"] = value
+            elif name == "MinKNOW":
+                info_dict["software"]["MinKNOW"] = value
+            elif name == "Bream":
+                info_dict["software"]["Bream"] = value
+            elif name == "MinKNOW Core":
+                info_dict["software"]["MinKNOW Core"] = value
+            elif name == "Configuration":
+                info_dict["software"]["Configuration"] = value
+            elif name == "Dorado":
+                info_dict["software"]["Dorado"] = value
+
     if json_file:
         print("[green]Reading info from json[/green]")
         with open(json_file[0]) as f: # assume only 1 file
             jsondata = json.load(f)
-        info_dict["flowcell"] = jsondata['protocol_run_info']['meta_info']['tags']['flow cell']['string_value']
-        info_dict["kit"] = jsondata['protocol_run_info']['meta_info']['tags']['kit']['string_value']
+
+        if not "flowcell" in info_dict:
+            info_dict["flowcell"] = jsondata['protocol_run_info']['meta_info']['tags']['flow cell']['string_value']
+        if not "kit" in info_dict:
+            info_dict["kit"] = jsondata['protocol_run_info']['meta_info']['tags']['kit']['string_value']
+        # HTML file is not reporting barcoding, we need it from the json
         info_dict['barcoding'] = bool(jsondata['protocol_run_info']['meta_info']['tags']['barcoding']['bool_value'])
-        info_dict['model_def'] = jsondata['protocol_run_info']['meta_info']['tags']['default basecall model']['string_value']
+        
+        # check run parameters to capture model, check base calling and alignment
+        # model is better defined in json, in html is badly reported
+        model = None
+        info_dict['do_basecall'] = 'do_basecall'
```

Comment:
Should this go in the config.yaml ?

Reply:
After seeing this used in the rules down the line, I guess more appropriate would be to infer this -> pod5's only -> do_basecall ?

Reply:
yes, but I am more into never do basecalling as that take forever, I will refactor the logic here